staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go

```diff
@@ -119,20 +128,36 @@ func UserInfo(namespace, name, uid string) user.Info {
 type ServiceAccountInfo struct {
 	Name, Namespace, UID string
 	PodName, PodUID      string
+	JTI                  string
+	NodeName, NodeUID    string
 }
 
 func (sa *ServiceAccountInfo) UserInfo() user.Info {
 	info := &user.DefaultInfo{
 		Name:   MakeUsername(sa.Namespace, sa.Name),
 		UID:    sa.UID,
 		Groups: MakeGroupNames(sa.Namespace),
+		Extra:  map[string][]string{},
 	}
+
 	if sa.PodName != "" && sa.PodUID != "" {
```

Comment:
slight preference for allocating Extra if needed and nil in each block, rather than eagerly allocating and then clearing:

```go
if info.Extra == nil {
  info.Extra = map[string][]string{}
}
```